umbra-js/src/classes/Umbra.ts

```diff
@@ -759,8 +770,10 @@ async function recursiveGraphFetch(
 
   // If there were results on this page then query the next page, otherwise return the data
   /* eslint-disable @typescript-eslint/no-unsafe-return */
-  if (!json.data[key].length) return [...before];
-  else return await recursiveGraphFetch(url, key, query, [...before, ...json.data[key]], overrides);
+  if (json.data[key].length) {
+    yield json.data[key]; // yield the data for this page
+    yield* recursiveGraphFetch(url, key, query, [...before, ...json.data[key]], overrides); // yield the data for the next pages
+  }
   /* eslint-enable @typescript-eslint/no-unsafe-return */
```

Comment:
Was the `else` block intentionally dropped here? Seems like a bug may have been introduced. Also now that we use `yield`, maybe the eslint hint
```suggestion
  // If there were results on this page then query the next page, otherwise return the data
  if (json.data[key].length) yield json.data[key]; // yield the data for this page
  else yield* recursiveGraphFetch(url, key, query, [...before, ...json.data[key]], overrides); // yield the data for the next pages
``` is outdated?

Reply:
So I played around and the suggestion code is yielding the first page first result and immediately stopping.
I think the block of code
```
  if (json.data[key].length) {
    yield json.data[key]; // yield the data for this page
    yield* recursiveGraphFetch(url, key, query, [...before, ...json.data[key]], overrides); // yield the data for the next pages
  }
  ```
  is correct as it would return the latest call results and then call recursively from where it left off. And if it finds no data, then do nothing because all previous data has already been returned.

Reply:
Updated comments: https://github.com/ScopeLift/umbra-protocol/pull/558/commits/619eff1c918bd837a11b4a21fe6a5c13aeed19fa

Reply:
> So I played around and the suggestion code is yielding the first page result and immediately stopping.

Oh maybe because there was a typo in my suggestion? You can see the original code was:
```typescript
if (!json.data[key].length) return [...before];
else return recursiveGraphFetch(...);
```

But this PR flips the first condition to remove the `!`. I'm mainly just surprised that changing from `return` to `yield` results in a differing conditional here which is why I'm making sure I understand what's going on.

In the old approach, nothing was returned until we hit the `if (!...)` condition, but in the new approach return data is basically streamed since this is a generator, which is why we don't need that anymore?

Reply:
> return data is basically streamed since this is a generator, which is why we don't need that anymore?

I believe this is the reason. It's pretty much like


```
  if (!json.data[key].length) {
    return;
  }
  yield [...before, ...json.data[key]];
  yield* recursiveGraphFetch(url, key, query, [...before, ...json.data[key]], overrides);
  ```

Reply:
Yep make sense, thanks a lot!